lib/components/user/user-account-screen.js

```diff
@@ -0,0 +1,560 @@
+          <FormControl type='number' placeholder='_ _ _ _ _ _' />
+        </FormGroup>
+      </div>
+    )
+
+    // In theory there could be multiple home or work locations.
+    // Just pick the first one,
+    const homeLocation = savedLocations.find(loc => loc.type === 'home') || {
+      address: null,
+      icon: 'home',
+      type: 'home'
+    }
+    const workLocation = savedLocations.find(loc => loc.type === 'work') || {
+      address: null,
+      icon: 'briefcase',
+      type: 'work'
+    }
+
+    // Build an 'effective' locations list for display.
+    const effectiveLocations = [
+      homeLocation,
+      workLocation,
+      ...savedLocations.filter(loc => loc.type !== 'home' && loc.type !== 'work')
+    ]
+
+    const placesPane = (
+      <div>
+        <ControlLabel>Add the places you frequent often to save time planning trips:</ControlLabel>
+
+        {effectiveLocations.map((loc, index) => (
+          <FormGroup key={index}>
+            <InputGroup>
+              <StyledAddon title={loc.type}>
+                <FontAwesome name={loc.icon} />
+              </StyledAddon>
+              <FormControl
+                onChange={e => this._handleAddressChange(loc, e)}
+                placeholder={`Add ${loc.type}`}
+                type='text'
+                value={loc.address} />
+            </InputGroup>
+          </FormGroup>
+        ))}
+
+        {/* For adding a location. */}
+        <FormGroup>
+          <InputGroup>
+            <NewPlaceAddon>
+              <FontAwesome name='plus' />
+            </NewPlaceAddon>
+            <NewPlaceFormControl
+              onBlur={this._handleAddNewLocation}
+              placeholder='Add another place'
+              type='text'
+            />
+          </InputGroup>
+        </FormGroup>
+      </div>
+    )
+
+    const finishPane = (
+      <div>
+        <p>You are ready to start planning your trips.</p>
+      </div>
+    )
+
+    const panesForNewAccount = {
+      terms: { title: 'Create a new account', pane: termsOfUsePane, nextId: 'notifications', disableNext: !hasConsentedToTerms },
+      notifications: { title: 'Notification preferences', pane: notificationPrefsPane, nextId: notificationChannel === 'sms' ? 'verifyPhone' : 'places', prevId: 'terms' },
+      verifyPhone: { title: 'Verify your phone', pane: phoneVerificationPane, nextId: 'places', prevId: 'notifications', disableNext: true },
+      places: { title: 'Add your locations', pane: placesPane, nextId: 'finish', prevId: 'notifications' },
+      finish: { title: 'Account setup complete!', pane: finishPane, prevId: 'places' }
+    }
+    const realActivePane = panesForNewAccount[activePaneId]
+
+    const panesForExistingAccount = [
+      {title: 'Terms', pane: termsOfUsePane},
+      {title: 'Notifications', pane: notificationPrefsPane},
+      {title: 'My locations', pane: placesPane}
+    ]
+
+    return (
+      <form>
+        {isNewAccount
+          ? (
```

Comment:
To improve readability, I recommend breaking up each suboption into its own render function. Ex:

```js
<form>
  {isNewAccount ? this._renderNewAccount() : this._renderExistingAccount()}
</form>
```